x/ibc/04-channel/client/cli/query.go

```diff
@@ -70,3 +132,143 @@ func GetCmdQueryChannelClientState(clientCtx client.Context) *cobra.Command {
 	}
 	return cmd
 }
+
+// GetCmdQueryPacketCommitments defines the command to query all packet commitments associated with
+// a channel
+func GetCmdQueryPacketCommitments(clientCtx client.Context) *cobra.Command {
+	return &cobra.Command{
+		Use:     "packet-commitments [port-id] [channel-id]",
+		Short:   "Query all packet commitments associated with a channel",
+		Long:    "Query all packet commitments associated with a channel",
+		Example: fmt.Sprintf("%s query %s %s packet-commitments [port-id] [channel-id]", version.ClientName, host.ModuleName, types.SubModuleName),
+		Args:    cobra.ExactArgs(2),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			clientCtx = clientCtx.Init()
+			queryClient := types.NewQueryClient(clientCtx)
+
+			req := &types.QueryPacketCommitmentsRequest{
+				PortID:    args[0],
+				ChannelID: args[1],
+				Req:       &query.PageRequest{},
+			}
+
+			res, err := queryClient.PacketCommitments(context.Background(), req)
+			if err != nil {
+				return err
+			}
+
+			clientCtx = clientCtx.WithHeight(res.Height)
+			// TODO: return res?
+			return clientCtx.PrintOutput(res.Commitments)
+		},
+	}
+}
+
+// GetCmdQueryPacketCommitment defines the command to query a channel end
+func GetCmdQueryPacketCommitment(clientCtx client.Context) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "packet-commitment [port-id] [channel-id] [sequence]",
+		Short: "Query a packet commitment",
+		Long:  "Query a packet commitment",
+		Example: fmt.Sprintf(
+			"%s query %s %s end [port-id] [channel-id]", version.ClientName, host.ModuleName, types.SubModuleName,
+		),
+		Args: cobra.ExactArgs(3),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			clientCtx = clientCtx.Init()
+
+			portID := args[0]
+			channelID := args[1]
+			prove := viper.GetBool(flags.FlagProve)
+
+			seq, err := strconv.ParseUint(args[2], 10, 64)
+			if err != nil {
+				return err
+			}
+
+			res, err := utils.QueryPacketCommitment(clientCtx, portID, channelID, seq, prove)
+			if err != nil {
+				return err
+			}
+
+			clientCtx = clientCtx.WithHeight(int64(res.ProofHeight))
+			return clientCtx.PrintOutput(res)
+		},
+	}
+	cmd.Flags().Bool(flags.FlagProve, true, "show proofs for the query results")
+	return cmd
+}
+
+// GetCmdQueryUnrelayedPackets defines the command to query all the unrelayed packets.
+func GetCmdQueryUnrelayedPackets(clientCtx client.Context) *cobra.Command {
```

Comment:
I'd be nice if sequences could be an optional arg here, since most of the time the sequences is just the packet commitments queried for

Reply:
I added it as a flag now